flask_monitor/metrics.py

```diff
@@ -79,27 +87,61 @@ def after_request(response):
         # pylint: enable=protected-access
         error_status = is_error(response.status_code)
         METRICS_REQUEST_LATENCY \
-            .labels("http", response.status_code, error_status, request.method, request.path) \
+            .labels("http", response.status_code, error_status, "", request.method, request.path) \
             .observe(request_latency)
         METRICS_REQUEST_SIZE.labels(
-            "http", response.status_code, error_status, request.method, request.path
+            "http", response.status_code, error_status, "", request.method, request.path
         ).inc(size_request)
         return response
+
     if error_fn is not None:
         is_error.__code__ = error_fn.__code__
     app.before_request(before_request)
     app.after_request(after_request)
+    return app
 
-def watch_dependencies(dependency, func, time_execution=1500):
+def watch_dependencies(dependency, func, time_execution=1500, registry=None, app=current_app):
     """
     Register dependencies metrics
     """
+
+    # pylint: disable=invalid-name
+    DEPENDENCY_UP = Gauge(
+        'dependency_up',
+        'records if a dependency is up or down. 1 for up, 0 for down',
+        ["name"],
+        registry=registry
+    )
+
+    # pylint: disable=invalid-name
+    DEPENDENCY_UP_LATENCY = Histogram(
+        "dependency_request_seconds",
+        "records in a histogram the number of requests to dependency",
+        ["name", "type", "status", "isError", "errorMessage", "method", "addr"],
+        registry=registry
+    )
     def thread_function():
+        start = time.time()
         thread = threading.Timer(time_execution, lambda x: x + 1, args=(1,))
         thread.start()
         thread.join()
         response = func()
         DEPENDENCY_UP.labels(dependency).set(response)
+        DEPENDENCY_UP_LATENCY \
+            .labels(dependency, "http", "200", "False", "", "GET", "/") \
+            .observe(time.time() - start)
```

Comment:
@Ziul, why are you assuming `200`? shouldn't you wait for the result of `func`to know if the dependency is alive or not?

Reply:
Fixed the `200` and the `False`, using the `func` result.